### ISSUE TITLE
[DS-3708] Isolate DSpace-generated DOIs from other identifiers

### ISSUE THREAD

**Issue by dspace-bot** [DSpace/DSpace#7055]
<i>Imported from JIRA <a href="https://jira.lyrasis.org/browse/DS-3708">[DS-3708]</a> created by mwood</i>
<p>After discussion in Slack:#dev-mtg it is proposed that we move all DOIs generated by DSpace into a new field 'dc.identifier.doi'.  This will resolve the discrepancy between the DataCite and EZID DOI generation code (<a href="https://jira.lyrasis.org/browse/DS-2199" title="DSpace stores DOIs in different metadata values depending on which DOI registration agency is used" class="issue-link" data-issue-key="DS-2199"><del>DS-2199</del></a>), disentangle generated DOIs from Handles, and consistently separate generated DOIs from user-entered DOIs.  Thereafter, dc.identifier will hold all user-supplied identifiers; dc.identifier.doi will hold all DSpace-generated DOIs; and dc.identifier.uri will hold all Handles.  This will make storage of identifiers more consistent, prevent confusion of generated and submitted identifiers, and make it simpler to work with generated identifiers since each field will hold only one type.</p>

<p>Manually submitted DOIs will not be affected.</p>

<p>The patch should include code, such as a Flyway migration, to automatically resettle all generated identifiers held by existing DSpace instances.  DataCite DOIs can be recognized by being in dc.identifier.uri and not being Handles.  EZID DOIs can be recognized by being in dc.identifier and containing the prefix (value of identifier.doi.ezid.shoulder) that is configured for DOI generation.  DOIIdentifierProvider.supports(String) might be used to recognize DOIs regardless of representation.</p>

**Comment by bram-atmire** [comment]
This is still an issue today. The default that dc.identifier.uri is used, is pretty hard coded into the code here:

https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/identifier/DOIIdentifierProvider.java#L71-L73

At the very least, this should be pulled up into configuration so repository managers can more easily decide and change to store DOIs in a different field.

It creates all sorts of issues that repositories have both handles and DOIs thrown together in dc.identifier.uri.
Repository managers want to be able to make a conscious choice of which ONE single consistent identifier/permalink that they want to promote.

So if you're turning on the DOI integration, you see two links in dc.identifier.uri and it gets confusing for users what to use, cite ...

Anyway, all of this has already been said and discussed in this thread before. I'm just stating that in 2023, this issue still seems to persist.

**Comment by tdonohue** [comment]
Seems related or duplicative of #5565 , which has a (very old) possible fix in #1006 ... however it appears that fix has the opposite approach of moving everything to `dc.identifier.uri` away from `dc.identiifer.doi`.   So, we may need agreement on what field these DOIs should go into, or make it configurable.

**Comment by mwoodiupui** [comment]
My concern was to separate *identifiers under DSpace's control* from *identifiers under humans' control*.  But this issue raises a good separate point:  the reasonable desire to separate the various types of durable identifiers and (configurably) select the types which are presented.  It's time to have a discussion about how identifiers are entered, used, stored and presented in DSpace, and this is just one part of that discussion.  (This intersects the discussion about removing DSpace's ironclad dependence on Handles.)

I would have no objection to separating DSpace-controlled DOIs from DSpace-controlled Handles.  But I think we should consider namespaces too.  We have shoved several concepts into "dc" that are not defined by DCMI.  A redesign of identifier handling would be an opportunity to move those to a namespace which does define them, or to move them to the "dspace" namespace and formally define them ourselves.

**Comment by tdonohue** [comment]
@mwoodiupui : I'd recommend we take a staged approach here.... I worry if we wait to fully discuss a better approach for identifier handling, then the current buggy behavior could persist longer than we hope.

I'd consider the current behavior in this issue ticket o be a bug.  Two identifiers (DOI and Handle) are placed in the same `dc.identifier.uri` field, when our own UI expects that field to have a *single*, primary identifier.  Currently, our UI code simply grabs the first value in that field and displays it on the Community/Collection/Item page. So, on sites that use DOIs, this field might sometimes display a Handle and other times a DOI, depending on the first value listed.  And there's no way for a site to ensure that, globally, all their pages will have the same type of identifier.

So, I wonder if it'd be better to take a staged approach here:
1. First, ensure that `dc.identifier.uri` only contains a single Identifier type as that is how it has been used in v6 and now v7.  This is most easily done by "reserving" this field to the Handle (for now, just because that's the historical usage).
2. Second, move all DOIs (regardless of provider) to the existing `dc.identifier.doi` field.  This field already exists in our registry and is being used by some sites.  (We even had a question about using it for DOIs in our 7.6 Q&A webinar yesterday)
3. Once this first two points are solved, the bugs in both #5565 and this ticket will be solved.  But as you note, we likely then need to have a discussion about how to make this work better in the future.  Maybe we need to have a way to easily configure which metadata field to display as the primary identifier in the UI. Or, maybe define `dc.identifier.uri` as always being the "primary URI" field, but allow a way to configure which identifier is placed in that field (on the backend).  I agree, this seems like it does require more discussion and whether we even want to use the "dc" namespace for identifiers at all.

I'm just worried that if we wait on the third point...then the first two fixes (which are practically completed in your #1006 PR) will wait until this discussion completes.

**Comment by mwoodiupui** [comment]
I would point out that DC does not define `identifier.doi`.

**Comment by tdonohue** [comment]
Closing this bug ticket as it has been replaced by #9901 

This new ticket documents the DOI consensus achieved during the discussion in today's developers meeting (see [DevMtg on Oct 17, 2024](https://wiki.lyrasis.org/display/DSPACE/2024-10-17+DSpace+Developers+Meeting))
